airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -109,22 +109,29 @@ def read(
         with create_timer(self.name) as timer:
             for configured_stream in catalog.streams:
                 stream_instance = stream_instances.get(configured_stream.stream.name)
-                if not stream_instance:
-                    if not self.raise_exception_on_missing_stream:
-                        continue
+                is_stream_exist = bool(stream_instance)
+                try:
+                    # Used direct reference to `stream_instance` instead of `is_stream_exist` to avoid mypy type checking errors
```

Comment:
This comment clarifies why we do not use `is_stream_exist` instead of stream_instance in the ` if not stream_instance:`. The issue is that mypy will return an error if we use `is_stream_exist` instead of `stream_instance` in the condition.